schema/annotation-source.yaml

```diff
@@ -147,7 +152,17 @@ $defs:
       An InformationEntity representing an individual piece of data, generated/acquired through methods 
       which reliably produce truthful information about something.
     properties:
-      value:
+      type:
+        type: string
+        const: DataItem
+        default: DataItem
+        description: Must be "DataItem"
+     data_item_type: 
```

Comment:
I am having some difficulty with this pattern. I understand the rationale–we do not want to explicitly enumerate all data types. But I have a strong dislike of two "type" fields in an object, especially since this boils down to a free-text description of the data, and `description` is a property of `InformationEntity` (and is therefore inherited by `DataItem`) that seems to fit this purpose.

Reply:
I'll defer to you guys. This seems like a path we are heading down with Agent and Contribution (hopefully). But I'm not using this yet so I'm not invested in the decision at this point.

Reply:
We define a general pattern in the VA model there classes have an official 'type' (value must be a class from the VA model), and then a second optional attribute to allow specification of a more specific type where useful.  You utilize this pattern a couple times in the model already, e.g.:

````
  type: Agent
  agent_type: [ person, organization, software agent ]

  type: Method
  method_type:  [Variant Interpretation Guideline, Experimental Protocol, . . . ] 
````

If the concern has to do with the 'string' data type - I think we do want this to be an enumeration or Coding, and would be fine to change to whichever you think best here.  Hopefully this would alleviate your reticence.

Reply:
Would we be able to refer to this as a `subtype` instead of `data_item_type`, here and other places where this pattern is used? And yes, a `coding` would be more appropriate for subtypes in my view if this is something other than a `description` to orient the message receiver.

Reply:
Alex, is your proposal to rename all attributes implementing this pattern using the word 'subtype' instead of 'type'? e.g.:
- `agent_type` --> `agent_subtype`
- `method_type` --> `method_subtype`
- `contribution_type` --> `contribution_subtype`
- `data_item_type` --> `data_item_subtype`
- . . . 
 
Concrete examples of the difference:
 
````
type: Agent
agent_type:  Organization
 vs.
type: Agent
agent_subtype: Organization
````
 
````
type: Method
method_type:  Variant Interpretation Guideline
 vs.
type: Method
method_subtype:  Variant Interpretation Guideline
````
 
````
type: DataItem
data_item_type:  p-value
 vs.
type: DataItem
data_item_subtype:  p-value

Reply:
At the end of the day it's a minor difference and I could go either way, but I would prefer we keep it as is for now and see if it causes any issues with uses.  

For me, I am just not keen on against seeing the 'subtype' in these attribute names.  I've seen/used the general `"type": "X", "x_type": "X1"` pattern and naming approach in many other models and to me it is pretty clear. It didn't seem to raise any hackles on other classes (agent, method, etc) . . . is there something about data item that makes it problematic here? 

Whatever naming pattern we choose, I'm fine to change the value taken by  `data_item_type` to a Coding or enum as Alex suggests.

Reply:
This is a bit of a minor thing, but the problem is once you make it then you have to live with it, so it may be worth getting it right (with a solid justification).
By going with "subtype" we avoid the confusion with the "type" field.  Contribution.type vs Contribution.contribution_type is confusing on its face. Professionally, I try not to repeat the class name in the attribute names, because every attribute is inherently qualified by the class containing it. 
I can sort of understand how calling an attribute of X an X_<thing> you are really saying that this is a more specific level of the X instance. But has not been how I interpret class names. When I see an attribute with the classname as a prefix, I assume it is an unintentional redundancy.  There have been folks that put the datatype in the attribute names of variables and it has been discouraged as a beginner mistake for the most part (at least in "coding improvement" books and articles I've read over the years). 

I don't mean to blow up any big patterns you've settled on, but some reflection on those attributes that have the class  name in them might be worthwhile to determine if the additional redundant qualifier is meaningful.  Could it be an ontology related need that attribute names are thought of as independent entities and therefore need further specialization? 

In any case, we are trying to devise a schema and a model. and the model should clearly convey that any of the attributes of a component is named as a dependent characteristic of its parent. Thus there should not be any automatic need to repeat the classname in the name of the attribute.

In the end, if we change "type" to "subtype" it is a bit clearer that we are representing the X.subtype of the X.type (and the X.type would always be "X"), so there's a clear difference in the attributes and it should be super clear that subtype is a specialization of type.

Reply:
Thanks for this Larry - I have a clearer sense of your concerns. You guys are closer to the implementation/developer side, so if you prefer a different pattern, I will concede.
 
We could go with 'x_subtype' as Alex suggested.
 
We might also consider 'x_category' . . . 'category' is what FHIR uses when it wants to specify a more specific type (e.g. see  [Substance](https://www.hl7.org/fhir/substance.html), [ResearchStudy](https://www.hl7.org/fhir/researchstudy.html), and [Condition](https://www.hl7.org/fhir/condition.html)).  Not sure I love this however, as I generally consider the notion of a 'category' to be a broader/higher level bucket - and thus not entirely suited for the more granular typing we are trying to do here.

Reply:
i agree with your concerns over the use of category for this attribute. I think FHIR got it wrong. We should use 'subtype'.  There will be plenty of folks in the future to comment and provide input, we can change as the model matures.
Remember that we will presenting these classes as "Draft" or "Trial Use" which means they are experimental and subject to change.